src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -657,7 +675,7 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                             return task;
                         }
 
-                        goto case State.ResourceEnd;
+                        goto case State.InvokeEnd;
```

Comment:
Notice no tests changed to confirm a bug was fixed here. Suggest backing this out as it seems unrelated.